native/bls_nif/src/lib.rs

```diff
@@ -28,4 +28,30 @@ fn sign<'env>(
     Ok(bytes_to_binary(env, &bytes))
 }
 
-rustler::init!("Elixir.Bls", [sign]);
+#[rustler::nif]
+fn fast_aggregate_verify<'env>(
+    public_keys: Vec<Binary>,
+    message: Binary,
+    signature: Binary,
+) -> Result<bool, String> {
+    let aggregate_sig = match AggregateSignature::deserialize(signature.as_slice()) {
+        Ok(aggregate_sig) => aggregate_sig,
+        Err(e) => return Err(format!("{:?}", e)),
+    };
+    let pubkeys_result = public_keys
+        .iter()
+        .map(|pkb| match PublicKey::deserialize(pkb.as_slice()) {
+            Ok(pk) => Ok(pk),
+            Err(e) => return Err(format!("{:?}", e)),
+        })
+        .collect::<Result<Vec<PublicKey>, _>>();
+    let pubkeys = match pubkeys_result {
+        Ok(pubkeys) => pubkeys,
+        Err(e) => return Err(format!("{:?}", e)),
+    };
```

Comment:
This can be done with a `map_err` and a `?`:

```suggestion
    let pubkeys = pubkeys_result.map_err(|e| format!("{:?}", e))?;
```

Reply:
thanks, you are right with a `map_err` is much better.